build.gradle

```diff
@@ -364,56 +371,54 @@ project(':iceberg-pig') {
-project(':iceberg-runtime') {
+project(':iceberg-spark-runtime') {
   apply plugin: 'com.github.johnrengelman.shadow'
 
-  configurations {
-    shadow
-    compileOnly.extendsFrom shadow
-  }
-
   shadowJar {
-    configurations = [project.configurations.shadow]
     from(project.sourceSets.main.output)
   }
 
   tasks.build.dependsOn tasks.shadowJar
   tasks.install.dependsOn tasks.shadowJar
   tasks.javadocJar.dependsOn tasks.shadowJar
 
-  dependencies {
-    shadow project(':iceberg-api')
-    shadow project(':iceberg-common')
-    shadow project(':iceberg-core')
-    shadow project(':iceberg-parquet')
-    shadow project(':iceberg-spark')
-    shadow project(':iceberg-pig')
-    shadow project(':iceberg-hive')
-
-    shadow "org.apache.avro:avro"
-    shadow "org.apache.parquet:parquet-avro"
+  configurations {
+    compile {
+      // included in Spark
+      exclude group: 'org.slf4j'
+      exclude group: 'org.apache.commons'
+      exclude group: 'commons-pool'
+      exclude group: 'commons-codec'
+      exclude group: 'org.xerial.snappy'
+      exclude group: 'javax.xml.bind'
+    }
   }
 
-  publishing {
-    publications {
-      withType(IvyPublication) {
-        configurations {
-          'shadow' { extend 'runtime' }
-        }
-      }
-    }
+  dependencies {
+    compile project(':iceberg-spark')
   }
 
   shadowJar {
     zip64 true
 
     // Relocate dependencies to avoid conflicts
-    // relocate 'com.google.common', 'com.netflix.bdp.shaded.com.google.common'
+    relocate 'com.google', 'org.apache.iceberg.shaded.com.google'
     relocate 'com.fasterxml', 'org.apache.iceberg.shaded.com.fasterxml'
+    relocate 'com.github.benmanes', 'org.apache.iceberg.shaded.com.github.benmanes'
+    relocate 'org.checkerframework', 'org.apache.iceberg.shaded.org.checkerframework'
     relocate 'org.apache.avro', 'org.apache.iceberg.shaded.org.apache.avro'
-    relocate 'org.apache.parquet.avro', 'org.apache.iceberg.shaded.org.apache.parquet.avro'
-
-    archiveName = "iceberg-runtime-${version}.${extension}"
+    relocate 'avro.shaded', 'org.apache.iceberg.shaded.org.apache.avro.shaded'
+    relocate 'com.thoughtworks.paranamer', 'org.apache.iceberg.shaded.com.thoughtworks.paranamer'
+    relocate 'org.apache.parquet', 'org.apache.iceberg.shaded.org.apache.parquet'
+    relocate 'shaded.parquet', 'org.apache.iceberg.shaded.org.apache.parquet.shaded'
+    // relocate Avro's jackson dependency to share parquet-jackson locations
+    relocate 'org.codehaus.jackson', 'org.apache.iceberg.shaded.org.apache.parquet.shaded'
+    relocate 'org.apache.orc', 'org.apache.iceberg.shaded.org.apache.orc'
+    relocate 'io.airlift', 'org.apache.iceberg.shaded.io.airlift'
+    relocate 'org.apache.hive', 'org.apache.iceberg.shaded.org.apache.hive'
+    relocate 'org.apache.hadoop.hive', 'org.apache.iceberg.shaded.org.apache.hadoop.hive'
+
+    archiveName = "iceberg-spark-runtime-${version}.${extension}"
```

Comment:
By default `shadowJar` will add the `-all` classifier to the archive name so the archiveName by default will come out to be "iceberg-spark-runtime-<version>.<extension>-all.jar.   This give a good signal IMO that this specific jar is a fat jar.

Reply:
We don't use the `-all` classifier because this project is intended to build just one shaded Jar. It isn't a project that produces both a shaded version and a non-shaded version.